pkg/connector/role.go

```diff
@@ -84,19 +129,18 @@ func (o *roleResourceType) Entitlements(
 	token *pagination.Token,
 ) ([]*v2.Entitlement, string, annotations.Annotations, error) {
 	var rv []*v2.Entitlement
-
 	role := standardRoleFromType(resource.Id.GetResource())
-
-	en := sdkEntitlement.NewAssignmentEntitlement(resource, "assigned",
-		sdkEntitlement.WithDisplayName(fmt.Sprintf("%s Role Member", role.Label)),
-		sdkEntitlement.WithDescription(fmt.Sprintf("Has the %s role in Okta", role.Label)),
-		sdkEntitlement.WithAnnotation(&v2.V1Identifier{
-			Id: V1MembershipEntitlementID(role.Type),
-		}),
-		sdkEntitlement.WithGrantableTo(resourceTypeUser),
-	)
-
-	rv = append(rv, en)
+	if role != nil {
+		en := sdkEntitlement.NewAssignmentEntitlement(resource, "assigned",
+			sdkEntitlement.WithDisplayName(fmt.Sprintf("%s Role Member", role.Label)),
+			sdkEntitlement.WithDescription(fmt.Sprintf("Has the %s role in Okta", role.Label)),
+			sdkEntitlement.WithAnnotation(&v2.V1Identifier{
+				Id: V1MembershipEntitlementID(role.Type),
+			}),
+			sdkEntitlement.WithGrantableTo(resourceTypeUser),
+		)
+		rv = append(rv, en)
+	}
```

Comment:
This skips syncing entitlements for custom roles, and therefore skips syncing grants for them.